core/src/main/java/com/dtolabs/rundeck/core/storage/AuthRundeckStorageTree.java

```diff
@@ -58,13 +58,23 @@ public AuthRundeckStorageTree(StorageTree storageTree, AuthStorageContextProvide
      *
      * @return true if authorized
      */
-    private boolean authorizedPath(AuthContext context, Path path, String action) {
-        Decision evaluate = context.evaluate(
-                resourceForPath(path),
-                action,
-                contextProvider.environmentForPath(path)
-        );
-        return evaluate.isAuthorized();
+    private boolean authorizedPath(AuthContext context, Path path, String action)
+    {
+        boolean authorized = false;
+        Set<Attribute> environments = contextProvider.environmentForPath(path);
+        for (Attribute env : environments)
+        {
+            Decision evaluate = context.evaluate(
+                    resourceForPath(path),
+                    action, Collections.singleton(env)
+
+            );
+            if (evaluate.isAuthorized()){
```

Comment:
The behavior should also check if any of the decision.explain().getCode() is a REJECTED_DENIED state. in that case, the access should be rejected even if the other decision is authorized